### ISSUE TITLE
Run-time un/loading DLL is crashing

### ISSUE THREAD

**Issue by Invander29** [microsoft/mimalloc#208]
Hello, when I use LoadLibrary (Windows 10, v141) to open dll file, everything works, when I unload it using FreeLibrary, nothing happens, but when application is exiting, it crashes. Dll is compiled with mimalloc-static and does nothing, just includes `mimalloc-new-delete.h`.

Example:
`int main() {`
`  const auto handle = LoadLibrary(L"Foo.dll");`
`  FreeLibrary(handle);`
`  return 0;  // <-- crash is here, not in FreeLibrary function`
`}`

**Comment by rob-turner** [comment]
Hi Invader29,  I ran into a similar issue after unloading a dll that was using the statically linked version of mimalloc.  In fact we have 3 instances of the static version of mimalloc linked (exe -> client.dll -> server.dll).  The issue on Windows was caused by the use FlsAlloc and FlsSetValue in mimalloc.  These functions combined was causing the process to hold onto callback functions that are implemented in the loaded dll, and when our exe was terminating (along with the thread-pool) the runtime tried to dispatch to the functions in the unloaded dlls and crash.  I put in a work around to track the threads and then clean up the threads on thread termination and clean up all on dll unload.  The custom implementation of our Alloc and SetValue are scoped to the DLL instead of to the process.  It would be nice if mimalloc supported setting these callbacks via the API.  I changed the mimalloc source to use two external undefined functions to provide the FlsAlloc and FlsSetValue functions to mimalloc and these functions are resolved at link time with our implementation.  A short term fix to address the issue ASAP for our team. :-) 

Take care and have a great day!
